drivers/npmx/npmx_shell.c

```diff
@@ -401,12 +401,37 @@ static int cmd_charger_module_set(const struct shell *shell, size_t argc, char *
 		return 0;
 	}
 
+	npmx_error_t err_code;
+
+	if ((charger_set == 1) && (mask == NPMX_CHARGER_MODULE_NTC_LIMITS_MASK)) {
+		npmx_adc_ntc_type_t type;
+		npmx_adc_t *adc_instance = npmx_adc_get(npmx_instance, 0);
+
+		err_code = npmx_adc_ntc_get(adc_instance, &type);
+
+		if (check_error_code(shell, err_code)) {
+			if (type == NPMX_ADC_NTC_TYPE_HI_Z) {
```

Comment:
It is not required to check whether NTC temperature limit control module was enabled after setting thermistor.

Reply:
This is the case when the thermistor type is not configured (hi_z) and the user wants to enable the ntc module.
So, in my opinion, it is absolutely necessary.

Reply:
Shell is not a place where those checks should be done. For example in BUCK, selecting a pin to enable buck, the pin should be checked whether is it selected as input. We decided not to add that logic to the shell, the user is responsible for the proper command order.

Reply:
I don't understand why this is not a good place? If something is not allowed to work, it is good practice to inform the user about a possible error.
Anyway, done as you wanted.